### PR TITLE
Handle statsd paths in different format

### PR DESCRIPTION
**Pull request by r1viollet**
# What does this PR do?

Add an environment variable to the internal stat field
Adjust the format if it contains `unix://`
Revert removal of env var for pprof_prefix

# Motivation

Make it easier to set the internal stats field.
Avoid breaking relenv
Fix prof-correctness tests

### CODE REVIEW

**Review by sanchda — APPROVED** [review]
Yeah, this is a good idea.  Thank you!

**Comment by r1viollet** [comment]
> Yeah, this is a good idea. Thank you!

Thanks. CI was failing due to documentation updates. I think PR is now ready!

File: include/statsd.hpp [hunk 1]
```diff
@@ -29,3 +29,5 @@ DDRes statsd_close(int);
 
 /* Private */
 DDRes statsd_listen(std::string_view path, int *fd);
+
+std::string_view adjust_uds_path(std::string_view path);
```
Comment (nsavoire) [inline include/statsd.hpp:33]: Does this need to be exposed in header ?

File: src/statsd.cc [hunk 2]
```diff
@@ -15,10 +15,18 @@
 #include <sys/un.h>
 #include <unistd.h>
 
+std::string_view adjust_uds_path(std::string_view path) {
+  // Check if the socket_path starts with unix://
+  if (path.substr(0, 7) == "unix://") {
+    path.remove_prefix(7);
+  }
+  return path;
```
Comment (nsavoire) [inline src/statsd.cc:23]:
```suggestion
  static constexpr std::string_view prefix{"unix://"};
  if (path.starts_with(prefix)) {
    path.remove_prefix(prefix.length());
  }
  return path;
```

**Review by nsavoire — DISMISSED** [review]

**Review by nsavoire — APPROVED** [review]
LGTM!

**Pull request merged**